landscape/client/broker/exchange.py

```diff
@@ -438,6 +439,20 @@ def _message_is_obsolete(self, message):
 
         return result
 
+    def truncate_message_field(self, field, message):
+        """
+        Truncates message field value based on length
+        """
+        if field in message:
+            value = message[field]
+            if isinstance(value, str):
```

Comment:
Is there a reason we are only truncating string fields? What about `bytes`? I wonder if it might be more valuable to add a `max_length` init parameter (with a default of `None`) to these schema classes https://github.com/CanonicalLtd/landscape-client/blob/master/landscape/lib/schema.py#L67-L102
and do the truncation on coercion.

Reply:
Those two fields being truncated are defined as Unicode() in the schema, so I added the extra check there to prevent trying to truncate a None value. 

If it's in the coerce, then I'm worried about truncating a json or something that is parsed. Those two fields I chose seem to only be for informational purposes

Reply:
Ok. If both fields are `Unicode`, and since we no longer support Python 2. All good.